The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13014668380/job/36300759454) [(plain)](https://github.com/rust-lang/rust/commit/a7f1ace42ce2777a374e439577aa7c8433334aec/checks/36300759454/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#21 exporting to docker image format
#21 sending tarball 27.7s done
#21 DONE 40.4s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
1 error[E0405]: cannot find trait `Equivalent` in this scope
-   --> $DIR/sysroot-private.rs:26:18
+   --> $DIR/sysroot-private.rs:27:18
3    |
4 LL | trait Trait2<K>: Equivalent<K> {}

6 
7 error[E0412]: cannot find type `K` in this scope
-   --> $DIR/sysroot-private.rs:31:35
-   --> $DIR/sysroot-private.rs:31:35
+   --> $DIR/sysroot-private.rs:32:35
9    |
10 LL | fn trait_member<T>(val: &T, key: &K) -> bool {

22    |                  +++
23 
24 error[E0220]: associated type `ExpressionStack` not found for `Trait`
24 error[E0220]: associated type `ExpressionStack` not found for `Trait`
-   --> $DIR/sysroot-private.rs:21:31
+   --> $DIR/sysroot-private.rs:22:31
26    |
27 LL | type AssociatedTy = dyn Trait<ExpressionStack = i32, Bar = i32>;
28    |                               ^^^^^^^^^^^^^^^ help: `Trait` has the following associated type: `Bar`
29 
30 error[E0425]: cannot find function `memchr2` in this scope
-   --> $DIR/sysroot-private.rs:39:5
+   --> $DIR/sysroot-private.rs:40:5
+   --> $DIR/sysroot-private.rs:40:5
32    |
33 LL |     memchr2(b'a', b'b', buf)

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/privacy/sysroot-private.rs:27:18
-   --> /checkout/tests/ui/privacy/sysroot-private.rs:32:35
---
To only update this specific test, also pass `--test-args privacy/sysroot-private.rs`

error in revision `default`: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/privacy/sysroot-private.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--cfg" "default" "--check-cfg" "cfg(test,FALSE,default,rustc_private_enabled)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/privacy/sysroot-private.default" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error[E0405]: cannot find trait `Equivalent` in this scope
##[error]  --> /checkout/tests/ui/privacy/sysroot-private.rs:27:18
   |
   |
LL | trait Trait2<K>: Equivalent<K> {}

error[E0412]: cannot find type `K` in this scope
##[error]  --> /checkout/tests/ui/privacy/sysroot-private.rs:32:35
   |
   |
LL | fn trait_member<T>(val: &T, key: &K) -> bool {
   |                 |
   |                 similarly named type parameter `T` defined here
   |
help: a type parameter with a similar name exists
help: a type parameter with a similar name exists
   |
LL | fn trait_member<T>(val: &T, key: &T) -> bool {
help: you might be missing a type parameter
   |
   |
LL | fn trait_member<T, K>(val: &T, key: &K) -> bool {

error[E0220]: associated type `ExpressionStack` not found for `Trait`
##[error]  --> /checkout/tests/ui/privacy/sysroot-private.rs:22:31
   |
   |
LL | type AssociatedTy = dyn Trait<ExpressionStack = i32, Bar = i32>;
   |                               ^^^^^^^^^^^^^^^ help: `Trait` has the following associated type: `Bar`
error[E0425]: cannot find function `memchr2` in this scope
##[error]  --> /checkout/tests/ui/privacy/sysroot-private.rs:40:5
   |
   |
LL |     memchr2(b'a', b'b', buf)

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0220, E0405, E0412, E0425.
---
1 error[E0405]: cannot find trait `Equivalent` in this scope
-   --> $DIR/sysroot-private.rs:26:18
+   --> $DIR/sysroot-private.rs:27:18
3    |
4 LL | trait Trait2<K>: Equivalent<K> {}

6 
7 error[E0412]: cannot find type `K` in this scope
-   --> $DIR/sysroot-private.rs:31:35
-   --> $DIR/sysroot-private.rs:31:35
+   --> $DIR/sysroot-private.rs:32:35
9    |
10 LL | fn trait_member<T>(val: &T, key: &K) -> bool {

22    |                  +++
23 
24 error[E0220]: associated type `ExpressionStack` not found for `Trait`
24 error[E0220]: associated type `ExpressionStack` not found for `Trait`
-   --> $DIR/sysroot-private.rs:21:31
+   --> $DIR/sysroot-private.rs:22:31
26    |
27 LL | type AssociatedTy = dyn Trait<ExpressionStack = i32, Bar = i32>;
28    |                               ^^^^^^^^^^^^^^^ there is an associated type `ExpressionStack` in the trait `gimli::read::op::EvaluationStorage`
29 
30 error[E0425]: cannot find function `memchr2` in this scope
-   --> $DIR/sysroot-private.rs:39:5
+   --> $DIR/sysroot-private.rs:40:5
+   --> $DIR/sysroot-private.rs:40:5
32    |
33 LL |     memchr2(b'a', b'b', buf)

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/privacy/sysroot-private.rs:27:18
-   --> /checkout/tests/ui/privacy/sysroot-private.rs:32:35
---
To only update this specific test, also pass `--test-args privacy/sysroot-private.rs`

error in revision `rustc_private_enabled`: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/privacy/sysroot-private.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--cfg" "rustc_private_enabled" "--check-cfg" "cfg(test,FALSE,default,rustc_private_enabled)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/privacy/sysroot-private.rustc_private_enabled" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error[E0405]: cannot find trait `Equivalent` in this scope
##[error]  --> /checkout/tests/ui/privacy/sysroot-private.rs:27:18
   |
   |
LL | trait Trait2<K>: Equivalent<K> {}

error[E0412]: cannot find type `K` in this scope
##[error]  --> /checkout/tests/ui/privacy/sysroot-private.rs:32:35
   |
   |
LL | fn trait_member<T>(val: &T, key: &K) -> bool {
   |                 |
   |                 similarly named type parameter `T` defined here
   |
help: a type parameter with a similar name exists
help: a type parameter with a similar name exists
   |
LL | fn trait_member<T>(val: &T, key: &T) -> bool {
help: you might be missing a type parameter
   |
   |
LL | fn trait_member<T, K>(val: &T, key: &K) -> bool {

error[E0220]: associated type `ExpressionStack` not found for `Trait`
##[error]  --> /checkout/tests/ui/privacy/sysroot-private.rs:22:31
   |
   |
LL | type AssociatedTy = dyn Trait<ExpressionStack = i32, Bar = i32>;
   |                               ^^^^^^^^^^^^^^^ there is an associated type `ExpressionStack` in the trait `gimli::read::op::EvaluationStorage`
error[E0425]: cannot find function `memchr2` in this scope
##[error]  --> /checkout/tests/ui/privacy/sysroot-private.rs:40:5
   |
   |
LL |     memchr2(b'a', b'b', buf)

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0220, E0405, E0412, E0425.
```

</details>

--- a/tests/ui/privacy/sysroot-private.default.stderr
+++ b/tests/ui/privacy/sysroot-private.default.stderr
@@ -1,11 +1,11 @@
 error[E0405]: cannot find trait `Equivalent` in this scope
-  --> $DIR/sysroot-private.rs:26:18
+  --> $DIR/sysroot-private.rs:27:18
    |
 LL | trait Trait2<K>: Equivalent<K> {}
    |                  ^^^^^^^^^^ not found in this scope
 
 error[E0412]: cannot find type `K` in this scope
-  --> $DIR/sysroot-private.rs:31:35
+  --> $DIR/sysroot-private.rs:32:35
    |
 LL | fn trait_member<T>(val: &T, key: &K) -> bool {
    |                 -                 ^
@@ -22,13 +22,13 @@ LL | fn trait_member<T, K>(val: &T, key: &K) -> bool {
    |                  +++
 
 error[E0220]: associated type `ExpressionStack` not found for `Trait`
-  --> $DIR/sysroot-private.rs:21:31
+  --> $DIR/sysroot-private.rs:22:31
    |
 LL | type AssociatedTy = dyn Trait<ExpressionStack = i32, Bar = i32>;
    |                               ^^^^^^^^^^^^^^^ help: `Trait` has the following associated type: `Bar`
 
 error[E0425]: cannot find function `memchr2` in this scope
-  --> $DIR/sysroot-private.rs:39:5
+  --> $DIR/sysroot-private.rs:40:5
    |
 LL |     memchr2(b'a', b'b', buf)
    |     ^^^^^^^ not found in this scope

--- a/tests/ui/privacy/sysroot-private.rs
+++ b/tests/ui/privacy/sysroot-private.rs
@@ -7,6 +7,7 @@
 //! of `std`'s dependencies, but may not be robust against dependency upgrades/changes.
 
 //@ only-unix Windows sysroots seem to not expose this dependency
+//@ ignore-emscripten neither does Emscripten
 //@ revisions: default rustc_private_enabled
 
 // Enabling `rustc_private` should `std`'s dependencies accessible, so they should show up

--- a/tests/ui/privacy/sysroot-private.rustc_private_enabled.stderr
+++ b/tests/ui/privacy/sysroot-private.rustc_private_enabled.stderr
@@ -1,11 +1,11 @@
 error[E0405]: cannot find trait `Equivalent` in this scope
-  --> $DIR/sysroot-private.rs:26:18
+  --> $DIR/sysroot-private.rs:27:18
    |
 LL | trait Trait2<K>: Equivalent<K> {}
    |                  ^^^^^^^^^^ not found in this scope
 
 error[E0412]: cannot find type `K` in this scope
-  --> $DIR/sysroot-private.rs:31:35
+  --> $DIR/sysroot-private.rs:32:35
    |
 LL | fn trait_member<T>(val: &T, key: &K) -> bool {
    |                 -                 ^
@@ -22,13 +22,13 @@ LL | fn trait_member<T, K>(val: &T, key: &K) -> bool {
    |                  +++
 
 error[E0220]: associated type `ExpressionStack` not found for `Trait`
-  --> $DIR/sysroot-private.rs:21:31
+  --> $DIR/sysroot-private.rs:22:31
    |
 LL | type AssociatedTy = dyn Trait<ExpressionStack = i32, Bar = i32>;
    |                               ^^^^^^^^^^^^^^^ there is an associated type `ExpressionStack` in the trait `gimli::read::op::EvaluationStorage`
 
 error[E0425]: cannot find function `memchr2` in this scope
-  --> $DIR/sysroot-private.rs:39:5
+  --> $DIR/sysroot-private.rs:40:5
    |
 LL |     memchr2(b'a', b'b', buf)
    |     ^^^^^^^ not found in this scope

